src/components/Integrations/BigQuery/ControlPanel/ControlPanel.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useState, useMemo, useCallback } from 'react';
+import { useIntl } from 'react-intl';
+import MultiInputContainer from './MultiInputContainer';
+import { FormattedMessageMarkdown } from '../../../../i18n/FormattedMessageMarkdown';
+import { VerticalLine } from './style/styles.js';
+import datastudiogif from '../../../../img/google-data-studio.gif';
+
+interface IControlPanelProps {
+  values?: string[];
+  onChange: (newValues: string[]) => void;
+}
+
+const ControlPanel: React.FunctionComponent<IControlPanelProps> = (props) => {
+  const intl = useIntl();
+  const i18 = useCallback(
+    (id: string, values?: any) => {
+      return intl.formatMessage({ id: id }, values);
+    },
+    [intl],
+  );
+
+  const [values, setValues] = useState<string[]>([]);
+  const [message, setMessage] = useState(false);
+  const [saved, setSaved] = useState(false);
+
+  const onClickHandler = useCallback(
+    (isSaved: boolean) => {
+      if (isSaved) {
+        setMessage(isSaved);
+        setSaved(isSaved);
+      } else {
+        setMessage(isSaved);
+        setSaved(isSaved);
+      }
+    },
+    [setMessage, setSaved],
+  );
+
+  const onClickBtnHandler = useCallback(() => {
+    setMessage(true);
+    setSaved(false);
+  }, [setMessage, setSaved, values]);
+
+  const MessageMemo = useMemo(() => {
+    let messageText;
+    if (saved) {
+      messageText = <p>{i18('big_query.plus_message_saved')}</p>;
+    } else {
+      messageText = <p>{i18('big_query.plus_message_remember')}</p>;
+    }
+
+    if (message === true) {
+      return (
+        <div className="p-t-96" data-testid="message-info">
+          <div className={`dp-wrap-message dp-wrap-${saved ? 'success' : 'info'}`}>
+            <span className="dp-message-icon"></span>
+            <div className="dp-content-message">{messageText}</div>
+          </div>
+        </div>
+      );
+    }
```

Comment:
It's not necessary `if(necessary === true)`. You can simply do: `if(message)` or `if(!message)`.

Reply:
remove data-testid, it's not necessary. There are other ways